Sources/PackageModel/Manifest/PackageDependencyDescription.swift

```diff
@@ -11,15 +11,33 @@
 import TSCBasic
 
 /// Represents a package dependency.
-public struct PackageDependencyDescription: Equatable, Codable, Hashable {
+public enum PackageDependencyDescription: Equatable {
+
+     public struct Local: Equatable, Codable {
+        public let identity: PackageIdentity
+        public let name: String?
+        public let path: AbsolutePath
+        public let productFilter: ProductFilter
+    }
+
+    public struct Git: Equatable, Codable {
+        public let identity: PackageIdentity
+        public let name: String?
+        public let location: String
```

Comment:
Is this always a URL in this case?  Would it make sense to use URL, or rather keep as String?